tuxedo-core/src/constraint_checker.rs

```diff
@@ -22,13 +122,16 @@ pub trait SimpleConstraintChecker: Debug + Encode + Decode + Clone {
     /// The error type that this constraint checker may return
     type Error: Debug;
 
+    /// A transient accumulator that can be used to track intermediate data during the course of a block's execution.
+    type Accumulator: Accumulator;
+
     /// The actual check validation logic
     fn check(
         &self,
         input_data: &[DynamicallyTypedData],
         peek_data: &[DynamicallyTypedData],
         output_data: &[DynamicallyTypedData],
-    ) -> Result<TransactionPriority, Self::Error>;
+    ) -> Result<ConstraintCheckingSuccess<<Self::Accumulator as Accumulator>::ValueType>, Self::Error>;
```

Comment:
I think type aliasing for stuff like this is preferred. Things can start to look really ugly without it..